File: player/ultraPlayer/SetDlsRequest.cpp
#include "SetDlsRequest.h"

SetDlsRequest::SetDlsRequest(void):ARequest(MessageType::SetDls)
{
}

SetDlsRequest::~SetDlsRequest(void)
{
}

std::string SetDlsRequest::DlsFile()
{
	return _dlsFile;
}

void SetDlsRequest::FromString( string data )
{
	try
	{
		int index = CommunicationProtocol::Instance()->HeaderLength();

		_dlsFile = data.substr(index, data.length() - index);
	}
	catch (exception& e)
	{

	}
}

File: controller/Model/Communication/Request/SetDlsRequest.cs
using System;
using System.Collections.Generic;
using System.Text;
using UltraPlayerController.Communication;
using UltraPlayerController.Model.Enumerations;

namespace UltraPlayerController.Model.Communication.Request
{
    public class SetDlsRequest : ARequest
    {
        #region Fields
        private string _dlsFile;
        #endregion

        #region Constructors
        public SetDlsRequest(): base(MessageType.SetDls)
        {

        }
        #endregion

        #region ARequest members
        public override string ParseToString()
        {
            string request = string.Empty;
            try
            {
                request += base.ParseToString();
                request += _dlsFile;
            }
            catch (System.Exception)
            {

            }
            return request;
        }
        #endregion

        #region Properties
        public string DlsFile
        {
            get { return _dlsFile; }
            set { _dlsFile = value; }
        }
        #endregion
    }
}
